pkg/controller/unifiedpushserver/unifiedpushserver_controller.go

```diff
@@ -375,70 +383,83 @@ func (r *ReconcileUnifiedPushServer) Reconcile(request reconcile.Request) (recon
 	} else if err != nil {
 		return reconcile.Result{}, err
 	}
+	//#endregion
 
-	// Define a new Deployment object
-	unifiedpushDeployment, err := newUnifiedPushServerDeployment(instance)
+	//#region OauthProxy ImageStream
+	oauthProxyImageStream, err := newOauthProxyImageStream(instance)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
 
-	if err := controllerutil.SetControllerReference(instance, unifiedpushDeployment, r.scheme); err != nil {
+	// Set UnifiedPushServer instance as the owner and controller
+	if err := controllerutil.SetControllerReference(instance, oauthProxyImageStream, r.scheme); err != nil {
 		return reconcile.Result{}, err
 	}
 
-	// Check if this Deployment already exists
-	foundUnifiedpushDeployment := &appsv1.Deployment{}
-	err = r.client.Get(context.TODO(), types.NamespacedName{Name: unifiedpushDeployment.Name, Namespace: unifiedpushDeployment.Namespace}, foundUnifiedpushDeployment)
+	// Check if this ImageStream already exists
+	foundOauthProxyImageStream := &imagev1.ImageStream{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: foundOauthProxyImageStream.Name, Namespace: oauthProxyImageStream.Namespace}, foundOauthProxyImageStream)
```

Comment:
I think this line here is the cause of the errors in the logs that @psturc noticed. the `NameSpacedName` here should be using `oauthProxyImageStream.Name` rather than `foundOauthProxyImageStream.Name`. I'll fix it in my next PR.

Reply:
+1